internal/pkg/earpointer/analysis.go

```diff
@@ -747,6 +812,30 @@ func (vis *visitor) unifyByValue(ref1 Reference, ref2 Reference) {
 	}
 }
 
+// Return all the calling contexts of the function to which a value belongs.
+func (vis *visitor) getContexts(v ssa.Value) []*Context {
+	if fn := v.Parent(); fn != nil {
+		if ctxs, ok := vis.contexts[fn]; ok {
+			return ctxs
+		}
+	}
+	// Otherwise return an empty context, e.g. for a Global.
+	return []*Context{&emptyContext}
+}
+
+// Return whether two contexts are equal up to K.
+func contextKEqual(c1 Context, c2 Context) bool {
+	if len(c1) < contextK || len(c2) < contextK {
```

Comment:
I don't understand this what this predicate is trying to capture.

If the context is 2, the context {f} is not contextKEqual to itself?